components/eamxx/src/physics/ml_correction/eamxx_ml_correction_process_interface.cpp

```diff
@@ -32,8 +34,17 @@ void MLCorrection::set_grids(
 
   // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and
   // interfaces
+  FieldLayout scalar2d_layout{ {COL}, {m_num_cols}};
   FieldLayout scalar3d_layout_mid{{COL, LEV}, {m_num_cols, m_num_levs}};
-  
+  FieldLayout horiz_wind_layout { {COL,CMP,LEV}, {m_num_cols,2,m_num_levs} };
+  if (not m_ML_correction_unit_test) {
```

Comment:
It is also used in the unit test input yaml. The unit test does not have access to surface geopotential or lat/lon, and since it adds `ml_correction` it goes through the init (not implementation) and crashes.

On second thought, do we have to register MLCorrection in the unit test?

Reply:
Okay I thought about this more, we definitely want to register MLCorrection. I tried experimenting removing `m_ML_correction_unit_test` but kept running into different issues. Essentially, we have to be able to allocate all the fields MLCorrection needs during initialization, which includes variables such as `lat`, `phis`, and radiation fluxes (not this PR, but the next PR with ML surface fluxes). I can make it work for `lat` and `phis` by using actual scream IC and topography file, but it means slower unit test (which is probably okay). But, we will need a lot of dependencies on radiation in the next PR when we need fields like `SW_flux_dn`.

One obvious solution is to dynamically update pybind arguments based on runtime option, for example, only ask for surface fluxes and pass them if we intend to ML correct for those fields. The problem is that at the pybind11 interface, we cannot dynamically pass different number of variables. The only way would be to split it into 3 separate python calls (`Tq`, `Tquv`, `Tquv+surface fluxes`) but this will have a significant performance hit. 

So I'm inclined to keep `m_ML_correction_unit_test` flag and have a simple/fast unit test.